### PR TITLE
read only mode

### PR DESCRIPTION
**Pull request by mmvsha**

### CODE REVIEW

File: src/components/MarketList/index.js [hunk 1]
```diff
@@ -93,13 +91,11 @@ class MarketList extends Component {
...
-          !isResolved && (
-            <div className="market__control">
-              <a href={`/#${resolveUrl}`} onClick={e => this.handleViewMarketResolve(e, resolveUrl)}>
-                Resolve
-              </a>
-            </div>
-          )}
+            !isResolved && (
+              <div className="market__control">
+                <a onClick={e => this.handleViewMarketResolve(e, resolveUrl)}>Resolve</a>
```
Comment (andre-meyer) [inline src/components/MarketList/index.js]: Don't we need a href for `<a>` to work semantically?

File: src/components/MarketList/index.js [hunk 2]
```diff
@@ -236,15 +232,26 @@ class MarketList extends Component {
...
               <div className="col-xs-10 col-xs-offset-1 col-sm-12 col-sm-offset-0">
-                <button
-                  type="button"
-                  onClick={this.handleCreateMarket}
-                  className="marketStats__control btn btn-default"
-                  disabled={!this.props.defaultAccount}
-                  title={this.props.defaultAccount ? '' : 'Please connect to an ethereum network to create a market'}
+                <Tooltip
+                  overlay="You need a wallet connected in order to create a market"
+                  trigger={this.props.defaultAccount ? [] : ['hover']}
```
Comment (andre-meyer) [inline src/components/MarketList/index.js]: Can't you remove the tooltip completely if we don't have an account?

File: src/components/MarketList/index.js [hunk 3]
```diff
@@ -236,15 +232,26 @@ class MarketList extends Component {
...
                 >
-                  Create Market
-                </button>
+                  <div
+                    className={cn({
+                      'marketStats__control--container': true,
+                      disabled: !this.props.defaultAccount,
```
Comment (andre-meyer) [inline src/components/MarketList/index.js]: Can we have a selector for this and change the name? Something like `hasWallet`

File: src/components/Transactions/index.js [hunk 4]
```diff
@@ -80,42 +77,58 @@ const renderTransaction = type => ({
...
+    )
+  }
+}
 
 Transactions.propTypes = {
-  runningTransactions: PropTypes.arrayOf(PropTypes.object),
+  changeUrl: PropTypes.func,
   completedTransactions: PropTypes.arrayOf(PropTypes.object),
+  currentAccount: PropTypes.string,
+  runningTransactions: PropTypes.arrayOf(PropTypes.object),
```
Comment (andre-meyer) [inline src/components/Transactions/index.js]: Can you make a transaction Propshape? (not super important rn)

File: src/components/MarketList/index.js (hunk 1 above)
Comment (mmvsha) [reply src/components/MarketList/index.js]: @andre-meyer hmm, there is an eslint warning about this:
`Visible, non-interactive elements should not have mouse or keyboard event listeners (jsx-a11y/no-static-element-interactions)`
Probably we need to change it

File: src/components/MarketList/index.js (hunk 2 above)
Comment (mmvsha) [reply src/components/MarketList/index.js]: @andre-meyer Why? It tells user that he needs to connect a wallet

File: src/components/MarketList/index.js (hunk 3 above)
Comment (mmvsha) [reply src/components/MarketList/index.js]: @andre-meyer Selector for what? We have selector for current account: ```defaultAccount: getCurrentAccount(state),```
Shall i just rename the prop?

Comment (andre-meyer) [reply src/components/MarketList/index.js]: Make a new selector that figures out if we have a wallet. Right now it should simply return false if we have no current account, then rename the prop

File: src/components/MarketList/index.js (hunk 2 above)
Comment (andre-meyer) [reply src/components/MarketList/index.js]: But you set the trigger to []?

Comment (andre-meyer) [reply src/components/MarketList/index.js]: Oh! I meant if the user does have an account. We don't need the tooltip right?

File: src/components/MarketList/index.js (hunk 1 above)
Comment (andre-meyer) [reply src/components/MarketList/index.js]: how about the good old `href="javascript:void(0)"`

File: src/components/MarketList/index.js [hunk 5]
```diff
@@ -204,6 +205,30 @@ class MarketList extends Component {
...
+        className={cn({
+          'marketStats__control--container': true,
+          disabled: !this.props.hasWallet,
+        })}
+      >
+        <button
+          type="button"
+          onClick={this.props.hasWallet ? this.handleCreateMarket : false}
+          className="marketStats__control btn btn-default"
+          disabled={!this.props.defaultAccount}
```
Comment (andre-meyer) [inline src/components/MarketList/index.js]: use hasWallet here too

File: src/components/Header/index.js [hunk 6]
```diff
@@ -9,21 +9,31 @@ import { upperFirst } from 'lodash'
...
+  currentBalance,
+  currentProvider,
+  getProviderIcon,
+  openConnectWalletModal,
+}) => (
   <div className="headerContainer">
     <div className="container">
       <div className="headerContainer__group headerContainer__group--logo">
-        <Link to="/">
+        <Link to={currentAccount ? '/' : '/markets/list'}>
```
Comment (andre-meyer) [inline src/components/Header/index.js]: can you use the hasWallet selector here too?

File: src/components/Header/index.js [hunk 7]
```diff
@@ -9,21 +9,31 @@ import { upperFirst } from 'lodash'
...
         </Link>
       </div>
       <div className="headerContainer__group headerContainer__group--left headerContainer__group--version">
         {version}
       </div>
       <div className="headerContainer__group headerContainer__group--left">
-        <Link to="/dashboard" activeClassName="headerContainer__navLink--active" className="headerContainer__navLink">
-          Dashboard
-        </Link>
+        {currentAccount && (
```
Comment (andre-meyer) [inline src/components/Header/index.js]: Here too?

File: src/components/MarketDetail/index.js [hunk 8]
```diff
@@ -145,8 +142,10 @@ class MarketDetail extends Component {
         })
       })
 
-    this.props.requestGasCost(GAS_COST.BUY_SHARES)
-    this.props.requestGasCost(GAS_COST.SELL_SHARES)
+    if (this.props.defaultAccount) {
```
Comment (andre-meyer) [inline src/components/MarketDetail/index.js]: Use `hasWallet` selector

File: src/components/Header/index.js [hunk 9]
```diff
@@ -41,16 +51,31 @@ const Header = ({ version, currentAccount, currentBalance, currentNetwork, curre
...
-        {currentAccount && currentProvider && (
-          <div className="headerContainer__account">
-            {currentNetwork && currentNetwork !== 'MAIN' && <span className="headerContainer__network--text">Network: {upperFirst(currentNetwork.toLowerCase())}</span>}
-            <DecimalValue value={currentBalance} className="headerContainer__account--text" />&nbsp;<span className="headerContainer__account--text">ETH</span>
-            <Identicon account={currentAccount} />
-          </div>
-        )}
-        {currentAccount && currentProvider && (
-          <ProviderIcon provider={currentProvider} />
+        {hasWallet && currentProvider && getProviderIcon(currentProvider)}
```
Comment (andre-meyer) [inline src/components/Header/index.js]: Why `getProviderIcon` here? Is this not an image path?

**Pull request merged**